app/models/school.rb

```diff
@@ -74,7 +77,7 @@ def delivery_partner_for(year)
   end
 
   def early_career_teacher_profiles_for(year)
-    early_career_teacher_profiles.joins(:cohort).where(cohort: { start_year: year })
+    participant_profiles.ects.where(cohort_id: Cohort.where(start_year: year).select(:id))
```

Comment:
Do we need to change to use `cohort_id` from `cohort` in that query?

Reply:
The new query does not perform join which in ActiveRecord is alwyas a massive win - `joins` does not perform preloading and will make things really complicated when this is combined with `includes` and conditions on joined table. It is best to avoid `joins` whenever possible.